src/Document.js

```diff
@@ -1,15 +1,199 @@
-const {HTMLElement, HTMLLoadableElement, Node} = require('./DOM');
+const parse5 = require('parse5');
+
+const DOM = require('./DOM');
+const {Event} = require('./Event');
+const GlobalContext = require('./GlobalContext');
 const symbols = require('./symbols');
+const utils = require('./utils');
+
+const _parseDocument = (s, options, window) => {
+  const ast = parse5.parse(s, {
+    sourceCodeLocationInfo: true,
+  });
+  ast.tagName = 'document';
+  return _parseDocumentAst(ast, options, window, true);
+};
+module.exports._parseDocument = _parseDocument;
+GlobalContext._parseDocument = _parseDocument;
+
+const _parseDocumentAst = (ast, options, window, uppercase) => {
```

Comment:
I guess that means the options arg is dead.

Reply:
We can pass it down later if it ends up getting used?

Reply:
I think it's attached to the `window`, which also comes in here.

I'll clean it up; not really related to this PR. 👍